### PR TITLE
[Frontend PDPD] Multi-output nodes support

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
 - *Previous solution https://github.com/nosovmik/openvino/pull/1 added support for single output multi-tensors nodes. The new approach resolves limitations for multi-output nodes.*

### Tickets:
 - *52249*

based on https://github.com/nosovmik/openvino/pull/4

### CODE REVIEW

File: ngraph/frontend/paddlepaddle/src/op/split.cpp [hunk 1]
```diff
@@ -14,25 +14,35 @@
...
         const auto& data = node.get_ng_input("X");
         auto dim = node.get_attribute<int32_t>("axis");
         // todo: 'num' can be list of values, in this case we should create VariadicSplit
         // todo: support VariadicSplit
         auto num_or_sections = node.get_attribute<int32_t>("num");
         auto axis = std::make_shared<Constant>(ngraph::element::i32, Shape{}, dim);
 
-        return std::make_shared<ngraph::opset6::Split>(data, axis, num_or_sections)->outputs();
+        NamedOutputs named_outputs;
+        const auto& split_outputs = std::make_shared<Split>(data, axis, num_or_sections)->outputs();
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/op/split.cpp:36]: still not feeling very confident with 'const auto&' here, maybe just 'auto' will be better. However, this code with 'const auto&' seems working on my machine

File: ngraph/frontend/paddlepaddle/src/op/split.cpp [hunk 2]
```diff
@@ -14,25 +14,35 @@
...
+        const auto& out_names = node.get_output_names();
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/op/split.cpp:37]: and same here, maybe using just 'auto' will be more understandable

File: ngraph/frontend/paddlepaddle/src/op/transpose2.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@ OutputVector transpose2 (const NodeContext& node) {
     PDPD_ASSERT(perm.size() == rank, "transpose2: axis size must equal to data rank!");
 
     auto input_order = ngraph::opset6::Constant::create(ngraph::element::i64, {rank}, perm);
-    return {std::make_shared<ngraph::opset6::Transpose>(data, input_order)};
+    return node.default_single_output_mapping({std::make_shared<ngraph::opset6::Transpose>(data, input_order)}, {"Out"});
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/op/transpose2.cpp:37]: should this be 'opset7' as well? Or it is planned to fix everywhere later?

File: ngraph/frontend/paddlepaddle/src/node_context.hpp [hunk 4]
```diff
@@ -70,6 +73,10 @@ class NodeContext
             return false;
         }
     }
+
+    std::vector<OutPortName> get_output_names() const { return node.get_output_names(); }
+    NamedOutputs default_single_output_mapping(const std::shared_ptr<Node> &ngraph_node,
+                                               const std::vector<OutPortName>& pdpd_out_names) const;
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/node_context.hpp:79]: minor: should it be renamed to 'required_pdpd_out_names' for consistency?

File: ngraph/frontend/paddlepaddle/src/frontend.cpp [hunk 5]
```diff
@@ -135,20 +134,20 @@ std::shared_ptr<Function>
...
-                        int idx = 0;
-                        for (const auto& port : name_to_ports.second) {
-                            const auto& var = port->getTargetTensorPDPD()->getDesc();
-                            nodes_dict[var->name()] = node->output(idx);
-                            idx++;
-                        }
+                const auto& named_outputs = pdpd::make_ng_node(nodes_dict, op_place, CREATORS_MAP);
+                // set layer name by the name of first output var
+                const auto& first_output_var = op_place->getOutputPorts().begin()->second.at(0)->getTargetTensorPDPD()->getDesc();
+                const auto& node = named_outputs.begin()->second[0].get_node_shared_ptr();
```
Comment (mvafin) [inline ngraph/frontend/paddlepaddle/src/frontend.cpp:140]: `named_outputs` can be empty, please add if statement here.

**Review by mvafin — CHANGES_REQUESTED** [review]

File: ngraph/frontend/paddlepaddle/src/op/transpose2.cpp (hunk 3 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/op/transpose2.cpp:37]: it's better to fix it everywhere, the general recommendation is to use the latest available opset

File: ngraph/frontend/paddlepaddle/src/frontend.cpp (hunk 5 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/frontend.cpp]: fixed

File: ngraph/frontend/paddlepaddle/src/node_context.hpp (hunk 4 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/node_context.hpp]: done

File: ngraph/frontend/paddlepaddle/src/op/transpose2.cpp (hunk 3 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/op/transpose2.cpp:37]: will be done in separate PR

File: ngraph/frontend/paddlepaddle/src/op/split.cpp (hunk 2 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/op/split.cpp]: fixed

File: ngraph/frontend/paddlepaddle/src/op/split.cpp (hunk 1 above)
Comment (itikhono) [reply ngraph/frontend/paddlepaddle/src/op/split.cpp]: fixed

**Review by nosovmik — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Pull request merged**